spacelift/data_saved_filters.go

```diff
@@ -0,0 +1,110 @@
+package spacelift
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal"
+	"github.com/spacelift-io/terraform-provider-spacelift/spacelift/internal/structs"
+)
+
+func dataSavedFilters() *schema.Resource {
+	return &schema.Resource{
+		Description: "" +
+			"`spacelift_saved_filters` can find all saved filters that have certain type or name",
+
+		ReadContext: dataFiltersRead,
+
+		Schema: map[string]*schema.Schema{
+			"filter_type": {
+				Type:        schema.TypeString,
+				Description: "filter type to look for",
+				Optional:    true,
+			},
+			"filter_name": {
+				Type:        schema.TypeString,
+				Description: "filter name to look for",
+				Optional:    true,
+			},
+			"filters": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeString,
+							Description: "Globally unique ID of the saved filter",
+							Computed:    true,
+						},
+						"is_public": {
+							Type:        schema.TypeBool,
+							Description: "Toggle whether the filter is public or not",
+							Computed:    true,
+						},
+						"created_by": {
+							Type:        schema.TypeString,
+							Description: "Login of the user who created the saved filter",
+							Computed:    true,
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Description: "Name of the filter",
+							Computed:    true,
+						},
+						"type": {
+							Type:        schema.TypeString,
+							Description: "Type describes the type of the filter. It is used to determine which view the filter is for",
+							Computed:    true,
+						},
+						"data": {
+							Type:        schema.TypeString,
+							Description: "Data is the JSON representation of the filter data",
+							Computed:    true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataFiltersRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var query struct {
+		Filters []structs.SavedFilter `graphql:"savedFilters()"`
```

Comment:
It looks like the API supports querying by type (`savedFilters(type: String)`). Maybe we should use that and only do the name filtering client-side?